hs_composite_resource/tests/test_composite_resource.py

```diff
@@ -705,10 +772,39 @@ def test_can_be_public_or_discoverable_with_no_aggregation(self):
         # at this point resource can be public or discoverable
         self.assertEqual(self.composite_resource.can_be_public_or_discoverable, True)
 
+    def test_cannot_be_published_with_refenced_url_single_file_aggregation(self):
+        """
+        test a composite resource that includes referenced url single file aggregation cannot be
+        published
+        """
+        self.create_composite_resource()
+        url_file_base_name = 'test_url'
+        ret_status, msg = add_reference_url_to_resource(self.user,
+                                                        self.composite_resource.short_id,
+                                                        self.valid_url, url_file_base_name,
+                                                        'data/contents')
+        self.assertEqual(ret_status, status.HTTP_200_OK)
+        # at this point resource can't be public or discoverable as some core metadata missing
+        self.assertFalse(self.composite_resource.can_be_public_or_discoverable)
+        missing_elements = self.composite_resource.metadata.get_required_missing_elements()
```

Comment:
I think we're going to need to communicate to the user that this resource can't be created because it has referenced content.

Reply:
@sblack-usu Good point. I have added this case as one potential reason that makes this resource not be able to be published in ValidationError() being raised.